### PR TITLE
Add claim file delete button and allow uploading file on Revision status

### PR DESCRIPTION
**Pull request by hobbitronics**
# Add:
- claim file delete button and endpoint function
# Fix:
- Allow uploading a file on Revision status
- Show correct banners for Revision status
image

image
image

### CODE REVIEW

File: src/pages/policies/[policyId]/claims/[claimId].svelte [hunk 1]
```diff
@@ -83,10 +84,14 @@ $: householdId = policy.household_id ? policy.household_id : ''
...
-$: showRevisionMessage = claim.status_reason && claimStatus === ClaimStatus.Revision
+$: needsRevision = claimStatus === ClaimStatus.Revision
+$: showRevisionMessage = claim.status_reason && needsRevision
+$: showPrimaryBanner = !(needsRevision && isCustomer($roleSelection))
+$: showSecondaryBanner = needsFile && isCustomer($roleSelection)
 
 $: needsReceipt = claimStatus === ClaimStatus.Receipt
-$: needsFile = needsReceipt || isEvidenceNeeded(claimItem, claimStatus)
+$: needsFile = needsReceipt || needsRevision || isEvidenceNeeded(claimItem, claimStatus)
+$: allowDelete = needsRevision || [ClaimStatus.Receipt, ClaimStatus.Draft].includes(claimStatus)
```
Comment (Schparky) [inline src/pages/policies/[policyId]/claims/[claimId].svelte:94]: looks like the `needsRevision` is redundant

Comment (Schparky) [reply src/pages/policies/[policyId]/claims/[claimId].svelte:94]: Actually, not redundant, but could be expressed as an additional status in the list

**Review by Schparky — APPROVED** [review]

File: src/pages/policies/[policyId]/claims/[claimId].svelte (hunk 1 above)
Comment (hobbitronics) [reply src/pages/policies/[policyId]/claims/[claimId].svelte:94]: Do you mean for 'allowDelete'? I don't follow.

Comment (Schparky) [reply src/pages/policies/[policyId]/claims/[claimId].svelte:94]: `$: allowDelete = [ClaimStatus.Receipt, ClaimStatus.Draft, ClaimStatus.Revision].includes(claimStatus)

Comment (Schparky) [reply src/pages/policies/[policyId]/claims/[claimId].svelte:94]: Not necessarily better though. I'm on the fence.

Comment (hobbitronics) [reply src/pages/policies/[policyId]/claims/[claimId].svelte:94]: yeah, I'm fine with either as well

**Pull request merged**